app/controllers/discussions_controller.rb

```diff
@@ -34,6 +34,7 @@ def upvote
   end
 
   def create
+    validate_discussions_permission!
```

Comment:
Also, prefer using the `before_action` hooks when implementing controller validations, since they allow to reuse that logic across multiple actions - it was not the case here, but it is good to keep consistency